src/pykeen/losses.py

```diff
@@ -870,21 +919,25 @@ def process_lcwa_scores(
         labels: FloatTensor,
         label_smoothing: float | None = None,
         num_entities: int | None = None,
+        weights: FloatTensor | None = None,
     ) -> FloatTensor:  # noqa: D102
         # Sanity check
+        self._raise_on_weights(weights)
         if label_smoothing:
             labels = apply_label_smoothing(
                 labels=labels,
                 epsilon=label_smoothing,
                 num_classes=num_entities,
             )
 
-        return self(predictions=predictions, labels=labels)
+        return self(x=predictions, target=labels, reduction=self.reduction)
 
     def forward(
         self,
-        predictions: FloatTensor,
-        labels: FloatTensor,
+        x: FloatTensor,
+        target: FloatTensor,
+        weight: FloatTensor | None = None,
+        reduction: TorchReductionMethod = "mean",
```

Comment:
is passing `reduction` necessary? doesn't it already get processed and saved in self._reduction_method?

Reply:
I see in the other places, you swapped it out for on-the-fly lookup

Reply:
Good catch - I was thinking about having a "generic" weighted method by passing `none` and doing the weighted average reduction in the base class. But it might be better to push that into concrete loss functions (where a more efficient implementation could be possible).